queries/rescript/highlights.scm

```diff
@@ -66,50 +91,83 @@
 [
  (parameter (value_identifier))
  (labeled_parameter (value_identifier))
-] @parameter
+] @variable.parameter
 
 ; single parameter with no parens
-(function parameter: (value_identifier) @parameter)
+(function parameter: (value_identifier) @variable.parameter)
 
 ; first-level descructuring (required for nvim-tree-sitter as it only matches direct
 ; children and the above patterns do not match destructuring patterns in NeoVim)
-(parameter (tuple_pattern (tuple_item_pattern (value_identifier) @parameter)))
-(parameter (array_pattern (value_identifier) @parameter))
-(parameter (record_pattern (value_identifier) @parameter))
+(parameter (tuple_pattern (tuple_item_pattern (value_identifier) @variable.parameter)))
+(parameter (array_pattern (value_identifier) @variable.parameter))
+(parameter (record_pattern (value_identifier) @variable.parameter))
+
+; function identifier in let binding
+(let_binding 
+  pattern: (value_identifier) @function
+  body: (function))
+
+; function calls
+
+(call_expression 
+  function: (value_identifier_path
+    _
+    (value_identifier) @function.call))
+
+(call_expression 
+  function: (value_identifier) @function.call)
+
+; highlight the right-hand side of a pipe operator as a function call
+(pipe_expression
+  _
+  [(value_identifier_path
+    _
+    (value_identifier) @function.call)
+    (value_identifier) @function.call])
+
 
 ; Meta
 ;-----
 
-(decorator_identifier) @annotation
+(decorator_identifier) @attribute
 
 (extension_identifier) @keyword
 ("%") @keyword
 
 ; Misc
 ;-----
 
-(subscript_expression index: (string) @property)
-(polyvar_type_pattern "#" @constant)
+(polyvar_type_pattern "#" @constructor)
+
+[
+  "include"
+  "open"
+] @keyword.import
+
+
+[
+   "as"
+   "private"
+   "mutable"
+   "rec"
+] @keyword.modifier
+
+[
+  "type"
+] @keyword.type
 
 [
-  ("include")
-  ("open")
-] @include
+  "and"
+  "with"
+] @keyword.operator
 
 [
-  "as"
   "export"
   "external"
   "let"
   "module"
-  "mutable"
-  "private"
-  "rec"
-  "type"
-  "and"
   "assert"
   "await"
```

Comment:
Would you mind moving this one to `@keyword.coroutine`?

Reply:
Sure, I moved both `async` and `await`